sound/soc/sof/intel/hda-stream.c

```diff
@@ -120,8 +120,14 @@ int hda_dsp_stream_setup_bdl(struct snd_sof_dev *sdev,
 
 	period_bytes = hstream->period_bytes;
 	dev_dbg(sdev->dev, "period_bytes:0x%x\n", period_bytes);
+	/*
+	 * HDA spec demands at least two 'periods' for the DMA to work correctly
+	 * Note: period_bytes == 0 can only happen for firmware or library
+	 * loading, the binary sizes have the necessary alignment requirements
+	 * met
+	 */
 	if (!period_bytes)
-		period_bytes = hstream->bufsize;
+		period_bytes = hstream->bufsize / 2;
```

Comment:
Sorry, I didn't get the relationship between the comment and the change. Why do we change period_bytes from hstream->bufsize to hstream->bufsize / 2?

Reply:
because then there'll be 2 period each of size bufsize/2

Reply:
Thanks for the explanation. I got it now.